views/settings/index.js

```diff
@@ -36,50 +43,81 @@ export default class SettingsView extends React.Component {
   }
 
   state = {
-    loggedIn: false,
-    success: false,
-    loading: false,
+    loggedInGoogle: false,
+    loggedInStOlaf: false,
+    successGoogle: false,
+    successStOlaf: false,
+    loadingGoogle: false,
+    loadingStOlaf: false,
     attempted: false,
+    username: null,
+    password: null,
   }
 
   componentWillMount() {
     this.loadData()
   }
 
   loadData = async () => {
-    let [status] = await Promise.all([
+    let [creds, status] = await Promise.all([
+      loadLoginCredentials(),
       AsyncStorage.getItem('credentials:valid').then(val => JSON.parse(val)),
     ])
+    if (creds) {
+      this.setState({username: creds.username, password:creds.password})
+    }
     if (status) {
-      this.setState({success: true})
+      this.setState({loggedInStOlaf: true})
     }
   }
 
-  logIn = () => {
+  logInGoogle = () => {
     this.props.navigator.push({
       id: 'SISLoginView',
       index: this.props.route.index + 1,
       props: {
-        onLoginComplete: status => this.setState({success: status}),
+        onLoginComplete: status => this.setState({successGoogle: status}),
       },
     })
   }
 
-  logOut = async () => {
-    this.setState({loading: true})
+  logOutGoogle = async () => {
+    this.setState({loadingGoogle: true})
     CookieManager.clearAll((err, res) => {
       if (err) {
         console.log(err)
       }
       console.log(res)
       this.setState({
-        success: false,
-        loading: false,
+        successGoogle: false,
+        loadingGoogle: false,
       })
     })
     await AsyncStorage.setItem('credentials:valid', JSON.stringify(false))
   }
 
+  logInStOlaf = async () => {
+    this.setState({loadingStOlaf: true})
+    let {username, password} = this.state
+    if (username) {
+      username.replace('@stolaf.edu', '') // just in case someone uses their full email, throw the part we don't need away
```

Comment:
Might as well just make this generic; remove anything after `@`? Something like `.replace(/@.*$/, '')`?